### ISSUE TITLE
find_decl hangs the web prover

### ISSUE THREAD

**Issue by htzh** [leanprover/tutorial#95]
If one runs the following example from chapter 5:
```
import data.nat

open nat
find_decl ((_ * _) = (_ * _))
find_decl (_ * _) = _, +assoc
find_decl (_ * _) = _, -assoc

find_decl _ < succ _, +imp, -le
```
the prover just hangs. Furthermore no other example will run afterwards. I need to refresh to restart lean.js. This happens both on Chrome and Firefox. I checked that find_decl works on my local native emacs/lean installation so this appears to be an issue for the web version only.
This is with the newest build of the web version after https://github.com/leanprover/lean/issues/561 was resolved.

**Comment by leodemoura** [comment]
@soonhokong The problem is that the web version is running out of memory.
The example works if we use `?mem=64`.
In the past, we used to get an error message for "out of memory", this is not the case anymore.
It is also not clear to users how to increase the amount of memory.
The easy fix is to put the error message back and say something like "the memory can be increased by appending the option ?mem=<size> in the web address".
BTW, I increased the default to 64Mb (from 16Mb), but it has not propagated yet.
